program-runtime/src/compute_budget.rs

```diff
@@ -227,7 +232,9 @@ impl ComputeBudget {
         .unwrap_or(MAX_UNITS as u64)
         .min(MAX_UNITS as u64);
 
-        Ok(prioritization_fee.unwrap_or(0))
+        Ok(prioritization_fee
+            .map(|fee_type| PrioritizationFeeDetails::new(fee_type, self.max_units))
+            .unwrap_or_default())
```

Comment:
This returns a struct that can be used to calculate either the prioritization fee or prioritization fee rate correctly when handling either the new `SetPrioritizationFeeRate` instruction or the `RequestUnitsDeprecated` instruction when users specify an explicit fee.